template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/smm/StreamsMessagingManagerServiceConfigProvider.java

```diff
@@ -1,67 +1,78 @@
 import java.util.List;
 
 import org.springframework.stereotype.Component;
 
 import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
 import com.google.common.collect.Lists;
 import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil;
 import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
 import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRdsRoleConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
 import com.sequenceiq.cloudbreak.template.views.RdsView;
 
 @Component
 public class StreamsMessagingManagerServiceConfigProvider extends AbstractRdsRoleConfigProvider {
 
-    static final String DATABASE_TYPE = "smm_database_type";
+    public static final String DATABASE_TYPE = "smm_database_type";
 
-    static final String DATABASE_NAME = "smm_database_name";
+    public static final String DATABASE_NAME = "smm_database_name";
 
-    static final String DATABASE_HOST = "smm_database_host";
+    public static final String DATABASE_HOST = "smm_database_host";
 
-    static final String DATABASE_PORT = "smm_database_port";
+    public static final String DATABASE_PORT = "smm_database_port";
 
-    static final String DATABASE_USER = "smm_database_user";
+    public static final String DATABASE_USER = "smm_database_user";
 
-    static final String DATABASE_PASSWORD = "smm_database_password";
+    public static final String DATABASE_PASSWORD = "smm_database_password";
+
+    public static final String DATABASE_JDBC_URL_OVERRIDE = "database_jdbc_url_override";
 
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
-        final List<ApiClusterTemplateConfig> configList;
+        List<ApiClusterTemplateConfig> config = new ArrayList<>();
+        String cmVersion = ConfigUtils.getCmVersion(source);
         String cdhVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
                 "" : source.getBlueprintView().getProcessor().getStackVersion();
         if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_2_0)) {
             RdsView smmRdsView = getRdsView(source);
-            configList = List.of(
-                    config(DATABASE_TYPE, dataBaseTypeForCM(smmRdsView.getDatabaseVendor())),
-                    config(DATABASE_NAME, smmRdsView.getDatabaseName()),
-                    config(DATABASE_HOST, smmRdsView.getHost()),
-                    config(DATABASE_PORT, smmRdsView.getPort()),
-                    config(DATABASE_USER, smmRdsView.getConnectionUserName()),
-                    config(DATABASE_PASSWORD, smmRdsView.getConnectionPassword())
-            );
+            config.add(config(DATABASE_TYPE, dataBaseTypeForCM(smmRdsView.getDatabaseVendor())));
```

Comment:
Not a big deal but it would perhaps be sufficient to set `DATABASE_TYPE` in the `else` branch, i.e. when SSL is not enabled.